### PR TITLE
Examples of dynamic memory allocation.

### PR DESCRIPTION
**Pull request by salizarana**
* Example of malloc and realloc.
* Sum of n numbers from user input using calloc.
* Example of structure using dynamic memory allocation.

### CODE REVIEW

File: chapters/dynamic memory allocation/QN001.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Write a program to dynamically allocate memory to store a string and reallocate with higher 
+ * storage size and assign lengthier string and print both values before and after reallocation.
+ * Also free up the allocated.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN001.c:9]:
```suggestion
#include <string.h>

```
new line here

File: chapters/dynamic memory allocation/QN001.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#define NUll 0
+
+int main()
+{
+  char *string;
+
+  string = (char *)malloc(4 * sizeof(char));
+  if (string == NULL)
+  {
+    printf("Malloc failed\n");
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN001.c]:
```suggestion
    printf("Memory allocation failed\n");
```

File: chapters/dynamic memory allocation/QN001.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  printf("String of size %ld created.\n", sizeof(string));
+  strcpy(string, "Lele");
+  printf("String contains %s.\n", string);
+
+  string = realloc(string, 1 * sizeof(char));
+  if (string == NULL)
+  {
+    printf("Reallocation failed\n");
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN001.c]:
```suggestion
    printf("Memory reallocation failed\n");
```

File: chapters/dynamic memory allocation/QN001.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Write a program to dynamically allocate memory to store a string and reallocate with higher 
+ * storage size and assign lengthier string and print both values before and after reallocation.
+ * Also free up the allocated.
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN001.c]:
```suggestion
 * Also free up the allocated memory.
```

File: chapters/dynamic memory allocation/QN002.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  if (ptr == NULL)
+  {
+    printf("Memory cannot be allocated.\n");
+    exit(0);
+  }
+
+  printf("Enter array elements:\n");
+  for (i = 0; i < n; i++)
+  {
+    scanf("%d", &*(ptr + i));
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN002.c]:
```suggestion
    scanf("%d", (ptr + i));
```
Try this once, I suppose it works.

Comment (salizarana) [reply chapters/dynamic memory allocation/QN002.c]: Yes, it works. Thank you

File: chapters/dynamic memory allocation/QN003.c [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+ * Write a program to calculate exam percentage of n students using dynamic memory 
+ * allocation, function and structure. Also, free up the allocated space.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Subject
+{
+  int c_programming;
```
Comment (sarojrana) [inline chapters/dynamic memory allocation/QN003.c]:
```suggestion
  int cProgramming;
```

**Review by sarojrana — APPROVED** [review]

**Pull request merged**